### DISCUSSION TITLE
타입 스크립트 경험이 있다고 하셨는데, 어떤 이유로 사용해보셨나요?

### DISCUSSION THREAD

**Discussion by BaekYuri** [wanted-9th-3team/frontend-study#19] [프론트 기술 질문 공간]
타입 스크립트 경험이 있다고 하셨는데, 어떤 이유로 사용해보셨나요?

**Comment by michoball** [comment]
타입스크립트를 사용한 이유는 현재 프론트 엔드 개발의 트렌드이기도 하지만 사용해본 결과 자바스크립트로 개발 했을 때보다 많은 부분에서 장점을 가지고 있다고 생각해 최근 수행한 프로젝트에 모두 사용을 하였습니다. 

자바스크립트에 타입이 추가된 언어로서 모든 자바스크립트 프로젝트와 호환이 되고 함수나 변수등 다양한 부분에 타입을 지정해 역할을 명확하게 해서 코드 컴파일 단계에서 오류를 잡아낼 수 있습니다. 

런타입에서 에러를 잡아내기 전에 미리 에러를 잡아낼 수 있다는 점과 vscode와 같은 편집기에서 유용한 tooltip을 제공받을 수 있다는 것이 타입스크립트의 매우 큰 장점이라고 생각합니다 

또한 런타입에서도 타입 스크립트의 도움으로 어떤 종류의 에러가 어디서 났는지 보다 쉽게 알 수 있는 에러 메시지를 제공받아 디버깅을 용이하게 한다는 점도 큰 장점이라고 생각해 사용하게 되었습니다. 

이러한 장점들 때문에 지금은 거의 무조건 타입스크립트로 작업하게 되었습니다.

**Comment by javascriptbiggosoo** [comment]
타입스크립트가 할 수 있는 일은 모두 자바스크립트로도 가능합니다.
본래 자바스크립트 코딩이라면 방어코드 작성에서 주니어 개발자가 어려움을 겪을 수 있는데, 타입스크립트는 개발자가 특정 방식으로만 코딩하기를 강제합니다. 그렇기에 이에 맞춰 작성하기만 하면 어느정도 안정성을 보장해주는데 덕분에 전체적인 코드 품질의 상향 평준화가 된다는 효과가 있습니다.
타입을 지정하고나면 편집기에서 자동완성 기능을 제공해 편의성 측면에서도 이점이 있습니다.

**Comment by skyhanull** [comment]
타입스크립트의 가장 중요특징인 컴파일부분에서 오류를 잡아주는 것 때문에 사용하게 되었습니다
그리고 타입스크립트는 자바스크립트의 상위버전이라 둘을 같이 쓰는 데에 문제가 없기도 하고, 타입스크립트를 사용 하면 타입을 지정해 줄 수 있어 알아보기 편하다는 장점도 있습니다.
이러한 장점덕분에 타입스크립트를 사용하고 있습니다

**Comment by BaekYuri** [comment]
자바스크립트에 타입을 추가하면서, 생길 수 있는 오류를 사전에 방지할 수 있는 효과가 있다고 생각되어 사용하였습니다.
자바스크립트로 웹을 개발할 때, 객체의 속성에서 대소문자 오타와 같은 사소한 실수로 인해 원하는 대로 동작이 되지 않았던 경험이 있어 어려움을 겪었던 경험이 있습니다. 이를 타입 지정을 하면서, 어느 속성이 잘못 적혀있는지 코드를 작성하면서 확인할 수 있어 편리했습니다.

**Comment by bona373737** [comment]
graphql을 사용하여 react 프로젝트를 개발했던 경험이 있는데, 당시 프론트에서 필요한 데이터만을 선택적으로 요청할 수 있어서 프론트엔드 쪽에서 데이터를 편리하게 다뤄낼 수 있었습니다. 당시 느꼈던 아쉬운 점이 요청한 데이터의 타입이 명시되어 있지 않기 때문에 결국 api문서를 다시 찾아봐야 했던 점 이었습니다. 이 때의 경험으로 grapql과 함께 타입스크립트를 함께 사용했을때 프론트작업의 효율성이 올라갈 것이라 생각했습니다.
